app/src/main/java/com/codesoom/assignment/handler/TaskHttpHandler.java

```diff
@@ -0,0 +1,39 @@
+package com.codesoom.assignment.handler;
+
+
+import com.codesoom.assignment.handler.impl.*;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.*;
+import java.util.*;
+
+
+public class TaskHttpHandler implements HttpHandler {
+    Map<String, TaskHandler> handlerMap;
+
+    @Override
+    public void handle(HttpExchange exchange) throws RuntimeException {
+        initHandler(exchange);
+
+        handlerMap.entrySet().stream()
+                .filter(entry -> entry.getValue().isRequest())
+                .findFirst()
+                .ifPresent(entry -> {
+                    try {
+                        entry.getValue().handle();
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
```

Comment:
Optional의 값이 있는 경우를 하려고 하신 것 같습니다. 이런 경우는 안에서 다하는 것보다는 적절히 구분해주는게 좋아보이네요

```suggestion
        var entry = handlerMap.entrySet().stream()
                .filter(entry -> entry.getValue().isRequest())
                .findFirst();
       if (entry.isPresent()) {
           entry.getValue().handle();
       }
```

그리고 계속 예외를 잡고 다시 던지기 방법을 사용하고 계신 것 같은데, 특별한 이유가 없다면 에러는 한 번 만 잡는게 좋고, 잡아야 한다면 이유가 있어야 합니다. 여기서 단순히 예외를 바꿔서 던지는 것은 별로 좋아보이진 않아요

Reply:
람다식 내부에서 억지로 전부 끝내려고하다보니 저렇게 된것같습니다.
한번 내보내고 throw로 던져서 하나의 예외형태는 한군데서 처리하는게 기본적으로는 좋다는 말씀이시군요.
궁금한 점이 있습니다
에러를 중간에서 잡는다고한다면 어떤경우가 있을까요?
일단 제가 이해한것은 만약에 어떤 예외가 발생할 경우 뭔가 추가적으로
처리할 로직이 있는경우 중간에 잡아 주어야 한다는 걸까요?

Reply:
에러를 중간에서 잡는 경우는 사실 별로 없습니다. 더 큰 맥락을 알아야 에러를 처리할 수 있기 떄문이죠. 그래서 `에러를 한 번만 잡아라!` 라는 조언도 있습니다. 하지만 `에러를 잡고 다시 던지기`라는 것도 있습니다. 이것은 에러를 잡아서 다시 무언가를 바꿔서 던질려고 하는건데 예를들어 지금 경우가 `IOException`를 `RuntimeException`를 바꿔서 던진게 에러를 잡고 다시 던진거죠. 대부분 이런 경우는 에러 객체를 변경해주기 위해서 씁니다. 그런데 작성해주신 것을 보면 커스텀 에러 객체가 아니라 RuntimeException을 사용하고 있어서 다시 던지는 이유가 RuntimeException으로 바꾸는 이유인데 그게 납득이 되지는 않죠. 예를들어서 IOException 예외를 잡아서 EntryIsNotDefinedException으로 던진다면 의미를 더 부여하기 위해서 하는 것으로 볼 수 있겠죠

> 문제 발생 사실을 전달해야 하는 방향은 함수 호출의 역방향이다. 문제에 따라서는 더 큰 맥락을 알아야 문제를 해결하거나 대안을 제시할 수 있는데, 호출 사슬의 아래쪽보다 위쪽에 있는 함수들이 더 큰 맥락을 알고 있기 때문이다. - 핵심 코틀린 프로그래밍 - 5장 예외